components/collections/src/codepointtrie/cptrie.rs

```diff
@@ -861,6 +861,39 @@ impl<'trie, T: TrieValue> CodePointTrie<'trie, T> {
             .map(|cpm_range| cpm_range.range)
     }
 
+    /// Yields an [`Iterator`] returning the ranges of the code points whose values
+    /// match `predicate`.
+    ///
+    /// This is preferable to calling `.get_ranges().map()` since it will coalesce
+    /// adjacent ranges into one.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_collections::codepointtrie::planes;
+    ///
+    /// let trie = planes::get_planes_trie();
+    ///
+    /// let plane_val = 2;
+    /// let mut sip_range_iter = trie.get_ranges_mapped(|value| value != plane_val as u8).filter(|range| range.value);
+    ///
+    /// let end = plane_val * 0x1_0000 - 1;
+    ///
+    /// let sip_range = sip_range_iter.next()
+    ///     .expect("Complemented planes data should have at least one entry");
+    /// assert_eq!(0..=end, sip_range.range);
+    pub fn get_ranges_mapped<'a, U: Eq + 'a>(
```

Comment:
Observation: `get_ranges_for_value` and `get_ranges_mapped` should probably both be `iter_*` since the other function is called `iter_ranges` and we're returning iterators

Reply:
Yeah I agree, I was annoyed by this naming too. I'll fix the new one.